bears/vcs/git/GitCommitBear.py

```diff
@@ -151,6 +161,47 @@ def run(self,
             body,
             **self.get_issue_checks_metadata().filter_parameters(kwargs))
 
+    def check_revert(self, shortlog, body,
+                     allow_revert_commits_without_reason: bool = False,):
+        """
+        Validates revert commits.
+
+        :param shortlog:                 The shortlog message string.
+        :param body:          The body of the commit message of HEAD.
+        :param allow_revert_commits_without_reason: Whether reason is
+                                           needed for revert commits.
+        """
+        REVERT_REGEX = re.compile(r'(Revert ")([\S+\s*]*)("$)')
+        HEADLINE_REGEX = re.compile(r'(^This reverts commit )([0-9a-f]{40})')
+
+        if REVERT_REGEX.match(shortlog):
+            reverted_shortlog = re.search(REVERT_REGEX, shortlog).group(2)
+
+            body = body.strip('\n')
+            blankline = body.find('\n\n')
+            headline = body[:blankline] if blankline != -1 else body
+
+            if not HEADLINE_REGEX.match(headline):
+                yield Result(self, 'Invalid revert commit.')
+                return
+
+            commit_sha = re.search(HEADLINE_REGEX, headline).group(2)
+            command = ('git log -n 1 ' +
+                       str(commit_sha) + ' --pretty=%B')
+            rstdout, _ = run_shell_command(command)
```

Comment:
Question : What would happen if the commit_sha is an invalid number (or a commit_hash that doesn't exist) ?
[Refer](https://github.com/coala/coala-bears/pull/2241#issuecomment-376489510).